x-pack/plugins/serverless_search/common/services/decode_cloud_id.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// copied this solution from fleet due to time constraints
+// x-pack/fleet/common/services/decode_cloud_id.ts
+// decodeCloudId decodes the c.id into c.esURL and c.kibURL
+export function decodeCloudId(cid: string):
```

Comment:
this is the 2nd time we've copied this code, we should make a plan to abstract it to a package that can be shared between ent-search, serverless search and fleet instead of keeping it duplicated in 3 places.